bpfd-operator/controllers/bpfd-agent/bpfd-agent.go

```diff
@@ -506,3 +510,45 @@ func nodePredicate(nodeName string) predicate.Funcs {
 		},
 	}
 }
+
+func (r *BpfProgramReconciler) setNodeInterface(ourNode *v1.Node) error {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		r.Logger.Error(err, "failed to read node interfaces")
+		return err
+	}
+
+	for _, ipaddr := range ourNode.Status.Addresses {
+		r.Logger.V(1).Info("Node IP  - ", "Type", ipaddr.Type, "Address", ipaddr.Address)
+		if ipaddr.Type == v1.NodeInternalIP {
```

Comment:
It might make this logic a bit simpler to write this with `ForEachInterface` or `ForeachInterfaceAddress`

(https://github.com/tailscale/tailscale/blob/v1.38.3/net/interfaces/interfaces.go#L189) https://github.com/tailscale/tailscale/blob/v1.38.3/net/interfaces/interfaces.go#L231

Reply:
Wrote issue #385 come back and investigate.